### PR TITLE
Fixes: #3986 - Convert rack units to part of SVG rendered document

### LINKED ISSUE #3986: Printing a rack is not as expected
### Environment
* Python version:  3.6.9
* NetBox version:  2.7.2

Hi

Printing a rack on paper and/or pdf show devices under U1.
Here an example of a rack with version 2.7.2
rack-2 7 2

Here, the same rack with version 2.6.11
rack-2 6 11

### Steps to Reproduce
1. choose a rack with devices
2. print it on pdf (or use preview on MacOS)
3. All devices appears under the floor :)

### Expected Behavior
Devices are in the rack

### Observed Behavior
Devices are under the rack

### PR DESCRIPTION
**Pull request by DanSheps**
### Fixes: #3986 & #4110 

Converts the rack legend into part of the SVG rendered document

### CODE REVIEW

File: netbox/dcim/models/__init__.py [hunk 1]
```diff
@@ -435,9 +435,20 @@ def _draw_empty(drawing, rack, start, end, text, id_, face_id, class_, reservati
 
     def _draw_elevations(self, elevation, reserved_units, face, unit_width, unit_height):
 
-        drawing = self._setup_drawing(unit_width, unit_height * self.u_height)
+        # Setup vars for the rack unit
+        position_width = 30
```
Comment (jeremystretch) [inline netbox/dcim/models/__init__.py]: We should probably make this a keyword argument on `_draw_elevations()` to ensure flexibility. We can set it to 30 as a default, but it should be defined as a constant like we do with `RACK_ELEVATION_UNIT_WIDTH_DEFAULT` and `RACK_ELEVATION_UNIT_HEIGHT_DEFAULT`.

Also, what do you think about calling this something like `legend_width` instead? Position seems a bit confusing given the context.

File: netbox/dcim/models/__init__.py [hunk 2]
```diff
@@ -471,7 +482,7 @@ def _draw_elevations(self, elevation, reserved_units, face, unit_width, unit_hei
             unit_cursor += height
 
         # Wrap the drawing with a border
-        drawing.add(drawing.rect((0, 0), (unit_width, self.u_height * unit_height), class_='rack'))
+        drawing.add(drawing.rect((30, 0), (unit_width, self.u_height * unit_height), class_='rack'))
```
Comment (jeremystretch) [inline netbox/dcim/models/__init__.py]: `drawing.rect()` should reference a variable instead of `30`.

File: netbox/dcim/models/__init__.py [hunk 3]
```diff
@@ -435,9 +435,20 @@ def _draw_empty(drawing, rack, start, end, text, id_, face_id, class_, reservati
...
-        drawing = self._setup_drawing(unit_width, unit_height * self.u_height)
+        # Setup vars for the rack unit
+        position_width = 30
+
+        drawing = self._setup_drawing(unit_width + position_width, unit_height * self.u_height)
 
         unit_cursor = 0
+        for ru in range(0, self.u_height):
+            start_y = ru * unit_height
+            position_coordinates = (15, start_y + unit_height / 2 + 2)
```
Comment (jeremystretch) [inline netbox/dcim/models/__init__.py]: Where did the `15` come from?

File: netbox/dcim/models/__init__.py [hunk 4]
```diff
@@ -435,9 +435,20 @@ def _draw_empty(drawing, rack, start, end, text, id_, face_id, class_, reservati
...
+            unit = ru + 1 if self.desc_units else self.u_height - ru
+            drawing.add(
+                drawing.text("U" + str(unit), position_coordinates, class_="unit")
```
Comment (jeremystretch) [inline netbox/dcim/models/__init__.py]: We can omit the "U" from the unit number. (The current rendering does not include it.)

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/dcim/models/__init__.py (hunk 1 above)
Comment (DanSheps) [reply netbox/dcim/models/__init__.py]: Yeah, I think legend might be better.

Will do the constants ASAP

File: netbox/dcim/models/__init__.py (hunk 3 above)
Comment (DanSheps) [reply netbox/dcim/models/__init__.py]: Half of 30, since we are using a variable, changed it to that

**Review by jeremystretch — APPROVED** [review]

**Pull request merged**